src/utils/viewer/viewer.cpp

```diff
@@ -492,6 +506,33 @@ void Viewer::toggleNavMeshVisualization() {
   }
 }
 
+void Viewer::drawPickedObject() {
+  if (!objectPickingHelper_->objectPicked()) {
+    // nothing is picked, return directly
+    return;
+  }
+  // setup blending function
+  Mn::GL::Renderer::enable(Mn::GL::Renderer::Feature::Blending);
+  Mn::GL::Renderer::setBlendColor(Magnum::Color4{0.0, 0.0, 0.0, 0.5});
+  Mn::GL::Renderer::setBlendFunction(
+      Mn::GL::Renderer::BlendFunction::ConstantAlpha,
+      Mn::GL::Renderer::BlendFunction::OneMinusConstantAlpha);
```

Comment:
Why not have the selection / wireframe color with alpha set to `0.5f` instead? That then wouldn't need a special blend function, and thus two state switches less.